89.md

```diff
@@ -0,0 +1,62 @@
+NIP-89
+======
+
+Bitcoin Transaction Broadcasting Events
+-------------------------
+
+`draft` `optional` `author:benthecarman`
+
+A special event with kind `28333`, that contains an encoded bitcoin transaction.
+The bitcoin transaction is encoded as a base64 string, and is contained in the `content` field.
+
+The network magic is contained in the `tags` field. This is to identify the network that the transaction is for.
+The common ones are:
+
+- mainnet: f9beb4d9
+- testnet3: 0b110907
+- signet (default): 0a03cf40
+- regtest: fabfb5da
+
+If another value is present it is likely a custom signet.
+
+This is an ephemeral event, and is not stored on the server. So relays or other custom clients could receive these
+events, and broadcast them to the bitcoin network.
+
+For example:
+
+```json
+{
+  "kind": 28333,
+  "content": "AgAAAAABAtkXo8t2LfqIwQ+BPQg/LZdqAtAb61bY0zIjFvvxMTtGAAAAAAD9////j8wvOaFL5Dz+0CTKXNKxxv1GrusMps3aodsXQGp95xgAAAAAAP3///8CLAPTBQAAAAAiUSBIKoy1/JK1GdfWhii8KLfBooZg3qT7ImuwwwC6t2eT6oiKAQAAAAAAIlEgZtFI4Bx2rDqBe4X9+lzEhh7Kxca3tNE9l/QfH/ZghQECRzBEAiBV2O5+jskV8/H+UuzgRuku2gCAwlmErieXrS5scbongAIgLWD6uN8BGpyE7gNtX4jxnkNShY8yjhdcjhsHR4RAWJEBIQLlCnhSR+K9YoAngUWj94mptHhcVYJGi+rT3lwApIekkQJHMEQCIDwrK0YHftZ2BQsBjHdJb89Tlpzvc3z8E1wGgbqIVRBQAiB6J0UXuPjib+geVmaZeoq4vYD+fDz835o5OJRg9ITFDAEhA1oq2Zj0d4XbATjc3OUSXSJDvvIosDeAsHyegwk4fkcTNAslAA==",
+  "tags": [
+    [
+      "magic",
+      "0b110907"
+    ]
+  ],
+  ...other fields
+}
+```
+
+## Uses
+
+This offers a way for users to broadcast bitcoin transactions in a private way without having to connect to the bitcoin
+network directly.
+
+Today, users have to connect to the bitcoin network directly or use an api like mempool.space to broadcast transactions.
+This is not private, as you can leak your ip address to the service or a bitcoin node.
+
+Nostr provides a better opportunity for users to broadcast transactions in a private way. Nostr relays are at most
+bitcoin adjacent and have a much lower risk of being run by a chain analysis company, thus providing a possibly better
+way to broadcast transactions.
+
+## Recommendations
+
+Clients should generate an ephemeral keypair for each transaction they want to broadcast. This keypair should only ever
```

Comment:
What if people just used `G` as the key always instead of creating new keypairs all the time?

Reply:
Was thinking about using a static key, too.

But I think creating a new keypair (and deleting them afterwards) is more secure for individuals. With a known public keypair (with secret key also known since we need to sign events), there could be the chance that one may use this accidentally for other purposes. Like if the last used key is reused for something sensitive and then everyone else will be able to decrypt it.

However, not sure if this is a valid concern. But that I can even come up with something like this also tells me, that maybe the solution to just create keypairs and delete them is safer. Better safe than sorry in privacy/cybersecurity. I don't think it's too inconvenient to generate a new keypair every time to start using a less common mechanism for privacy.

Or are there other protocols which implement privacy by letting everyone use the same key?

Also, if I ever use this, I would go with the "generate my own keypairs for every event"-solution. So if enough people think like me, the people who use the same keypair will naturally be at a disadvantage (anonymity set becomes smaller and smaller). So they will be incentivized to generate their own keypairs, too.

So I think in the long run, `G` won't be used.

Reply:
> Since we can't ban people from not using G it could create a fingerprint of clients that use G and ones that don't.

Well said, that's basically what I wanted to say, too, haha